src/hera/workflows/parameter.py

```diff
@@ -25,6 +25,12 @@ class Parameter(_ModelParameter):
     output: Optional[bool] = False
     """used to specify parameter as an output in function signature annotations"""
 
+    dumper: Optional[Callable[[Any], str]] = None
+    """used to specify a dumper function to serialise the parameter value for Annotated parameters"""
+
+    loader: Optional[Callable[[str], Any]] = None
+    """used to specify a loader function to load the parameter value for Annotated parameters"""
```

Comment:
Could we document what exactly the `str` is going to be here? (e.g. will it be a path? loaders/dumpers typically write to/from disk, e.g. `json.dump`. or is it a string representation of an object? though I'd normally expect that to be called "serializer"/"deserializer".)

Reply:
Yeah on naming I was aiming for alignment with `json.loads/dumps` and Pydantic's `model_dump_json`, also `pickle.dumps`, therefore users would write `Parameter(..., dumper=pickle.dumps)`. I'd prefer not to use `serializer` and `deserializer` because they are long and look similar (from a distance...), which is probably also why other libraries use `dump`/`load`.